samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/OktaVerifyController.cs

```diff
@@ -0,0 +1,183 @@
+﻿
+
+namespace embedded_auth_with_sdk.Controllers
+{
+    using embedded_auth_with_sdk.Models;
+    using Microsoft.Owin.Security;
+    using Okta.Idx.Sdk;
+    using Okta.Idx.Sdk.OktaVerify;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Claims;
+    using System.Text;
+    using System.Threading.Tasks;
+    using System.Web.Mvc;
+
+    public class OktaVerifyController : Controller
+    {
+        private readonly IAuthenticationManager _authenticationManager;
+
+        private readonly IIdxClient _idxClient;
+
+        public OktaVerifyController(IAuthenticationManager authenticationManager, IIdxClient idxClient)
+        {
+            _authenticationManager = authenticationManager;
+            _idxClient = idxClient;
+        }
+
+        public ActionResult SelectAuthenticatorMethod()
+        {
+            var oktaVerifyAuthenticationOptions = (OktaVerifyAuthenticationOptions)Session[nameof(OktaVerifyAuthenticationOptions)];
+            return View(new OktaVerifySelectAuthenticatorMethodModel(oktaVerifyAuthenticationOptions));
+        }
+
+        public ActionResult Enroll()
+        {
+            var oktaVerifyEnrollOptions = (OktaVerifyEnrollOptions)Session[nameof(OktaVerifyEnrollOptions)];
+            return View("EnrollWithQrCode", new OktaVerifyEnrollPollModel(oktaVerifyEnrollOptions));
+        }
+
+        [HttpGet]
+        public ActionResult SelectEnrollmentChannel()
+        {
+            var oktaVerifyEnrollOptions = (OktaVerifyEnrollOptions)Session[nameof(OktaVerifyEnrollOptions)];
+            var selectEnrollmentChannelModel = new OktaVerifySelectEnrollmentChannelModel(oktaVerifyEnrollOptions);
+            return View(selectEnrollmentChannelModel);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> SelectEnrollmentChannel(OktaVerifySelectEnrollmentChannelModel model)
+        {
+            var oktaVerifyEnrollOptions = (OktaVerifyEnrollOptions)Session[nameof(OktaVerifyEnrollOptions)];
+            if (!ModelState.IsValid)
+            {
+                var selectEnrollmentChannelViewModel = new OktaVerifySelectEnrollmentChannelModel(oktaVerifyEnrollOptions)
+                {
+                    SelectedChannel = model.SelectedChannel,
+                };
+                return View(selectEnrollmentChannelViewModel);
+            }
+
+            await oktaVerifyEnrollOptions.SelectEnrollmentChannelAsync(model.SelectedChannel);
+
+            switch (model.SelectedChannel)
+            {
+                case "email":
+                    return View("EnrollWithEmail");
+                case "sms":
+                    return View("EnrollWithPhoneNumber", new OktaVerifyEnrollWithPhoneNumberModel { CountryCode = "+1" });
+            }
+
+            throw new ArgumentException($"Unrecognized Okta Verify channel: {model.SelectedChannel}");
```

Comment:
Nit: This should fail gracefully by showing the error in the sample app similarly to other scenarios.

Reply:
Good point.  Can you point me to an example of how that's currently handled?